src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public abstract class ManagedToolTask : ToolTask
```

Comment:
Glad you went with this abstraction. I think it simplifies this a bunch.

Reply:
Yeah. Figured it's what we're wanting msbuild to move to anyway 🙂